modules/standard/Sort.chpl

```diff
@@ -334,9 +340,11 @@ inline proc chpl_compare(a:?t, b:t, comparator:?rec) {
   // Use comparator.compare(a, b) if is defined by user
   } else if canResolveMethod(comparator, "compare", a, b) {
     return comparator.compare(a ,b);
-  } else if canResolveMethod(comparator, "keyPart", a, 0) {
+  } else if canResolveMethod(comparator, "chpl_keyPartInternal", a, 0) ||
+            canResolveMethod(comparator, "keyPart", a, 0) {
     return compareByPart(a, b, comparator);
   } else {
+    // TODO: this should talk about interfaces
```

Comment:
Did you want to address this TODO before merging or open an issue about it?

Reply:
I think we need `keyComparator` and `relativeComparator` before this can be resolved